File: BlueSTSDK/src/main/java/com/st/BlueSTSDK/Utils/advertise/BlueSTSDKAdvertiseInfo.java
package com.st.BlueSTSDK.Utils.advertise;

import com.st.BlueSTSDK.Node.Type;

public class BlueSTSDKAdvertiseInfo implements BleAdvertiseInfo {

    /**
     * device name
     */
    private final String mName;
    /**
     * device tx power
     */
    private final byte mTxPower;
    /**
     * device mac address
     */
    private final String mAddress;

    /**
     * bit map that tell us the available features
     */
    private final int mFeatureMap;
    /**
     * device id
     */
    private final byte mDeviceId;
    /**
     * Device ProtocolVersion (it is an unsigned char )
     */
    private final short mProtocolVersion;

    /**
     * board type -> is a super class of the device id
     */
    private final Type mBoardType;


    /**
     * board is in sleeping state
     */
    private final boolean mBoardSleeping;

    /**
     * board has general purpose info
     */
    private final boolean mHasGeneralPurpose;

    @Override
    public String getName() {
        return mName;
    }

    @Override
    public byte getTxPower() {
        return mTxPower;
    }

    @Override
    public String getAddress() {
        return mAddress;
    }

    @Override
    public int getFeatureMap() {
        return mFeatureMap;
    }

    @Override
    public byte getDeviceId() {
        return mDeviceId;
    }

    @Override
    public short getProtocolVersion() {
        return mProtocolVersion;
    }

    @Override
    public Type getBoardType() {
        return mBoardType;
    }

    @Override
    public boolean isBoardSleeping() {
        return mBoardSleeping;
    }

    @Override
    public boolean isHasGeneralPurpose() {
        return mHasGeneralPurpose;
    }

    public BlueSTSDKAdvertiseInfo(String mName, byte mTxPower, String mAddress, int mFeatureMap, byte mDeviceId, short mProtocolVersion, Type mBoardType, boolean mBoardSleeping, boolean mHasGeneralPurpose) {
        this.mName = mName;
        this.mTxPower = mTxPower;
        this.mAddress = mAddress;
        this.mFeatureMap = mFeatureMap;
        this.mDeviceId = mDeviceId;
        this.mProtocolVersion = mProtocolVersion;
        this.mBoardType = mBoardType;
        this.mBoardSleeping = mBoardSleeping;
        this.mHasGeneralPurpose = mHasGeneralPurpose;
    }


}

File: st_blue_sdk/src/main/java/com/st/blue_sdk/bt/advertise/BlueStSdkAdvertiseInfo.kt
/*
 * Copyright (c) 2022(-0001) STMicroelectronics.
 * All rights reserved.
 * This software is licensed under terms that can be found in the LICENSE file in
 * the root directory of this software component.
 * If no LICENSE file comes with this software, it is provided AS-IS.
 */
package com.st.blue_sdk.bt.advertise

import com.st.blue_sdk.models.Boards

data class BlueStSdkAdvertiseInfo(
    private val name: String,
    private val txPower: Byte,
    private val address: String?,
    /**
     * - for BlueSTSDK V1 -> bit map that tell us the available features
     * - for BlueSTSDK V2 -> Option bytes
     */
    private val optByteFeatureMap: Long,
    private val deviceId: Byte,
    private val protocolVersion: Short,
    private val model: Boards.Model,
    private val isSleeping: Boolean,
    private val hasGeneralPurpose: Boolean
) : BleAdvertiseInfo {

    override fun getName() = name

    override fun getTxPower() = txPower

    override fun getAddress() = address

    override fun getFeatureMap() = if (protocolVersion.toInt() == 1) optByteFeatureMap else 0

    override fun getOptionBytes() = if (protocolVersion.toInt() == 1) 0 else optByteFeatureMap

    override fun getDeviceId() = deviceId

    override fun getProtocolVersion() = protocolVersion

    override fun getBoardType() = model

    override fun isBoardSleeping() = isSleeping

    override fun hasGeneralPurpose() = hasGeneralPurpose
}